magento/magento2
# Usps add new allowed methods

### Description (*)

USPS made significant changes to its API since July 9 where it proposes a new service "USPS - Ground Advantage" instead of "First Class Package Service"

![image](https://github.com/magento/magento2/assets/115582581/be654ad9-fcbc-43dc-aa84-82b23faaa6e3)

So this PR is created in order to quickly activate the new shipping method proposed by USPS.

The deprecated methods can be used until September 2023.
Attached note sent by USPS
[USPS Grace Period.pdf](https://github.com/magento/magento2/files/12010384/USPS.Grace.Period.pdf)
[USPS Release Notes - Backwards Comp.pdf](https://github.com/magento/magento2/files/12010386/USPS.Release.Notes.-.Backwards.Comp.pdf)

Reference: 
API TOOL DOC: https://www.usps.com/business/web-tools-apis/2023-web-tools-release-notes.pdf

### Related Pull Requests


Reported Issue: [magento/magento2#37742]

### Fixed Issues (if relevant)

magento/magento2#37742
Related to MAGETWO-47504 commit: https://github.com/magento/magento2/commit/0932cd3cab435a1c4a051a7b5e4aefeef930d231

### Manual testing scenarios (*)


* You must have a test API in order to activate the USPS shipping method https://www.usps.com/business/web-tools-apis/#

1. From the backend > store > configuration > SALES > Delivery Methods > USPS > Allowed Methods
2. You will be able to select the new methods related to "USPS Ground Advantage"
3. With these methods selected and USPS enabled we can verify the frontend
4. Add to cart a product less than 70lbs in weight.
5. See in viewCart or Checkout, completing a real address, the active shipping methods should have "USPS Ground Advantage" among those enabled.

### Questions or comments


### Contribution checklist (*)
